src/main/host/host.rs

```diff
@@ -188,6 +188,10 @@ impl std::fmt::Debug for Host {
 }
 
 impl Host {
+    /// # Safety
+    ///
+    /// `dns` must be a valid pointer.
+    /// TODO: does this take ownership of `dns`?
```

Comment:
I think there is one global DNS object that is shared to all of the hosts. And since the DNS here is a C pointer, I assume that's still true.

Reply:
Fixed